src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -91,27 +93,35 @@ class EncodingShelfBase extends React.PureComponent<
             attachment="top left"
             targetAttachment="bottom left"
           >
-            {(fieldDef && !isWildcardChannelId(id) && contains(CUSTOMIZABLE_ENCODING_CHANNELS, id.channel)) ?
+            {(!isWildcardChannelId(id) && contains(CUSTOMIZABLE_ENCODING_CHANNELS, id.channel)) ?
```

Comment:
Since we don't allow customizing x and y valuedef, caret shouldn't be shown for x and y if there is no fieldDef.